### PR TITLE
Update pr-check.yaml

### PR DESCRIPTION
**Pull request by danivilla9**
Adds executing the E2E tests during the PR check, solves #315

### CODE REVIEW

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -117,4 +117,89 @@ jobs:
...
+
+      - name: Execute pnpm in OpenShift Local Extension
+        working-directory: ./crc-extension
+        run: pnpm install --no-frozen-lockfile
+
+      - name: Run All E2E tests
+        working-directory: ./crc-extension
+        env:
+          PODMAN_DESKTOP_ARGS: ${{ github.workspace }}/podman-desktop
+        run: pnpm test:e2e
```
Comment (ScrewTSW) [inline .github/workflows/pr-check.yaml:199]: Are you sure you want to run all the tests as part of prcheck? I'd reduce it to a smoke suit

**Review by ScrewTSW — COMMENTED** [review]
I'd reduce the suite scope for prcheck

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (odockal) [reply .github/workflows/pr-check.yaml:199]: right now we have just one test. so I think that adding smoke is quite premature.

**Review by odockal — APPROVED** [review]

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (odockal) [reply .github/workflows/pr-check.yaml:199]: but we will get to it.

**Review by odockal — CHANGES_REQUESTED** [review]
@danivilla9 PR seems ok. Have you tried it on your fork to check if it works?

File: .github/workflows/pr-check.yaml [hunk 2]
```diff
@@ -117,4 +117,89 @@ jobs:
...
+
+      - name: Ensure getting current HEAD version of the test framework
+        working-directory: ./crc-extension
+        run: |
+          # workaround for https://github.com/containers/podman-desktop-extension-bootc/issues/712
+          version=$(npm view @podman-desktop/tests-playwright@next version)
+          echo "Version of @podman-desktop/tests-playwright to be used: $version"
+          jq --arg version "$version" '.devDependencies."@podman-desktop/tests-playwright" = $version' package.json > package.json_tmp && mv package.json_tmp package.json
+
+      - name: Execute pnpm in OpenShift Local Extension
```
Comment (odockal) [inline .github/workflows/pr-check.yaml:191]: pnpm not yet supported. Use yarn until the switch is done.

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (odockal) [inline .github/workflows/pr-check.yaml:199]: Same as above.

**Review by odockal — CHANGES_REQUESTED** [review]

**Comment by danivilla9** [comment]
The new version has been tested successfully here: https://github.com/danivilla9/crc-extension/actions/runs/11256979671
The `yarn test` step at the end was removed because I think it was not needed. Can be brought back if I'm wrong.

File: .github/workflows/pr-check.yaml [hunk 3]
```diff
@@ -116,9 +116,6 @@ jobs:
       - name: Run formatter
         run: yarn format:check
 
-      - name: Run Tests
```
Comment (odockal) [inline .github/workflows/pr-check.yaml:119]: You should not remove unit tests ;)

File: .github/workflows/pr-check.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2022 Red Hat, Inc.
```
Comment (odockal) [inline .github/workflows/pr-check.yaml:2]: will be 2022-2024

File: .github/workflows/pr-check.yaml (hunk 3 above)
Comment (odockal) [reply .github/workflows/pr-check.yaml:119]: it is part of another job within this workflow.

**Review by odockal — CHANGES_REQUESTED** [review]

**Comment by odockal** [comment]
@danivilla9 we are getting closer. Take a look at comments and also rebase the branch, please.

**Comment by danivilla9** [comment]
The new changes have been tested successfully on my fork: https://github.com/danivilla9/crc-extension/actions/runs/11347317218/job/31558410384

File: .github/workflows/pr-check.yaml [hunk 5]
```diff
@@ -117,4 +117,78 @@ jobs:
...
+          mv tests/playwright/output/crc-tests-pd/plugins/extension/ tests/playwright/output/crc-tests-pd/plugins/crcextension
+          tar -xf /tmp/sso_extension.tar -C tests/playwright/output/crc-tests-pd/plugins/
+          mv tests/playwright/output/crc-tests-pd/plugins/extension/ tests/playwright/output/crc-tests-pd/plugins/ssoextension
+          export SKIP_INSTALLATION=true
+          
+      - name: Run All E2E tests
+        working-directory: ./crc-extension
+        env:
+          PODMAN_DESKTOP_ARGS: ${{ github.workspace }}/podman-desktop
+        run: yarn test:e2e
```
Comment (odockal) [inline .github/workflows/pr-check.yaml:194]: you are missing artifact upload action step.

**Review by odockal — CHANGES_REQUESTED** [review]

**Review by odockal — APPROVED** [review]

**Pull request merged**